e2e/testcases/cli_test.go

```diff
@@ -833,3 +850,206 @@ func TestNomosVetNamespaceRepo(t *testing.T) {
 		})
 	}
 }
+
+func TestCLIBugreportNomosRunningCorrectly(t *testing.T) {
+	nt := nomostest.New(t, nomostesting.NomosCLI, ntopts.RequireGKE(t))
+	tmpDir := nomostest.TestDir(t)
+	tw := nomostesting.New(t, nomostesting.NomosCLI)
+	var bugReportZipName, bugReportDirName string
+	_ = nomostest.NewOptStruct(nomostest.TestClusterName(tw), tmpDir, tw)
+	t.Cleanup(func() {
+		// remove zip and folder
+		_ = os.Remove(bugReportZipName)
+		_ = os.RemoveAll(bugReportDirName)
+	})
+
+	// Confirm all the namespaces and pods are up
+	time, e := nomostest.Retry(nt.DefaultReconcileTimeout, func() error {
+		var errs status.MultiError
+		errs = status.Append(errs, nt.Validate(KubeSystem, "", &corev1.Namespace{}))
+		errs = status.Append(errs, nt.Validate(configsync.ControllerNamespace, "", &corev1.Namespace{}))
+		if nt.MultiRepo {
+			errs = status.Append(errs, nt.Validate(configmanagement.RGControllerNamespace, "", &corev1.Namespace{}))
+			errs = status.Append(errs, nt.Validate(metrics.MonitoringNamespace, "", &corev1.Namespace{}))
+			errs = status.Append(errs, nt.Validate(core.RootReconcilerPrefix, configsync.ControllerNamespace, &appsv1.Deployment{}, nomostest.StatusEquals(nt, kstatus.CurrentStatus)))
+			errs = status.Append(errs, nt.Validate(configmanagement.RGControllerName, configmanagement.RGControllerNamespace, &appsv1.Deployment{}, nomostest.StatusEquals(nt, kstatus.CurrentStatus)))
+			errs = status.Append(errs, nt.Validate(reconcilermanager.ManagerName, configsync.ControllerNamespace, &appsv1.Deployment{}, nomostest.StatusEquals(nt, kstatus.CurrentStatus)))
+			errs = status.Append(errs, nt.Validate(metrics.OtelCollectorName, metrics.MonitoringNamespace, &appsv1.Deployment{}, nomostest.StatusEquals(nt, kstatus.CurrentStatus)))
+		} else {
+			errs = status.Append(errs, nt.Validate(filesystem.GitImporterName, configsync.ControllerNamespace, &appsv1.Deployment{}, nomostest.StatusEquals(nt, kstatus.CurrentStatus)))
+			errs = status.Append(errs, nt.Validate(state.MonitorName, configsync.ControllerNamespace, &appsv1.Deployment{}, nomostest.StatusEquals(nt, kstatus.CurrentStatus)))
+		}
+		return errs
+	})
+	if e != nil {
+		nt.T.Fatal("timed out waiting for resources to be ready")
+	}
+	nt.T.Logf("took %v to wait for resources to be ready", time)
+
+	// get bugreport
+	out, err := exec.Command("nomos", "bugreport").CombinedOutput()
+	if err != nil {
+		tw.Log(string(out))
+		tw.Fatal(err)
+	}
+
+	// locate zip file
+	bugReportZipName, err = getBugReportZipName("./", nt)
```

Comment:
It generates the bugreport in the current directory. If you move it to `nt.TmpDir`, you wouldn't need to delete it in the end. The test framework will blow away the entire `nt.TmpDir` after the test.

Reply:
Moving zip file and unzipping into nt.TmpDir